src/Storages/Cache/RemoteCacheController.cpp

```diff
@@ -0,0 +1,249 @@
+#include <Storages/Cache/RemoteCacheController.h>
+#include <Storages/Cache/ExternalDataSourceCache.h>
+#include <Storages/Cache/RemoteFileMetadataFactory.h>
+#include <IO/ReadBuffer.h>
+#include <IO/ReadSettings.h>
+#include <Poco/JSON/JSON.h>
+#include <Poco/JSON/Parser.h>
+#include <fstream>
+
+namespace DB
+{
+namespace fs = std::filesystem;
+namespace ErrorCodes
+{
+    extern const int OK;
+    extern const int BAD_ARGUMENTS;
+    extern const int LOGICAL_ERROR;
+    extern const int END_OF_FILE;
+}
+
+bool RemoteCacheController::loadInnerInformation(const fs::path & file_path)
```

Comment:
This does not really make sense as we can know return value of `loadInnerInformation` without even calling it because return value equals `fs::exists(file_path)`. I suggest to move this loading into constructor of `RemoteCacheController` and add check for path existance just before initializing object of this class.

Reply:
done